domains/certificates/Query.API/API/Program.cs

```diff
@@ -19,107 +22,126 @@
-var b2COptions = builder.Configuration.GetSection(B2COptions.Prefix).Get<B2COptions>()!;
-builder.Services.AddOptions<B2COptions>().BindConfiguration(B2COptions.Prefix).ValidateDataAnnotations()
-    .ValidateOnStart();
-builder.Services.AddB2CAndTokenValidation(b2COptions, tokenValidationOptions);
+        builder.Services.AddQueryApi();
+        builder.Services.AddContractService();
+        builder.Services.AddMeasurementsSyncer();
+        builder.Services.AddIssuingContractCleanup();
+        builder.Services.AddVersioningToApi();
 
+        var tokenValidationOptions =
+            builder.Configuration.GetSection(TokenValidationOptions.Prefix).Get<TokenValidationOptions>()!;
+        builder.Services.AddOptions<TokenValidationOptions>().BindConfiguration(TokenValidationOptions.Prefix)
+            .ValidateDataAnnotations().ValidateOnStart();
+        var b2COptions = builder.Configuration.GetSection(B2COptions.Prefix).Get<B2COptions>()!;
+        builder.Services.AddOptions<B2COptions>().BindConfiguration(B2COptions.Prefix).ValidateDataAnnotations()
+            .ValidateOnStart();
+        builder.Services.AddB2CAndTokenValidation(b2COptions, tokenValidationOptions);
 
-var app = builder.Build();
 
-app.MapHealthChecks("/health");
+        var app = builder.Build();
 
-app.AddSwagger("certificates");
+        app.MapHealthChecks("/health");
 
-app.UseHttpsRedirection();
+        app.AddSwagger("certificates");
 
-app.UseAuthentication();
-app.UseAuthorization();
+        app.UseHttpsRedirection();
 
-app.MapControllers();
+        app.UseAuthentication();
+        app.UseAuthorization();
 
-var activityLogApiVersionSet = app.NewApiVersionSet("activitylog").Build();
-app.UseActivityLog().WithApiVersionSet(activityLogApiVersionSet)
-    .HasApiVersion(ApiVersions.Version20240423AsInt)
-    .HasApiVersion(ApiVersions.Version20230101AsInt);
-app.UseActivityLogWithB2CSupport().WithApiVersionSet(activityLogApiVersionSet).HasApiVersion(ApiVersions.Version20240515AsInt);
+        app.MapControllers();
 
+        var activityLogApiVersionSet = app.NewApiVersionSet("activitylog").Build();
+        app.UseActivityLog().WithApiVersionSet(activityLogApiVersionSet)
+            .HasApiVersion(ApiVersions.Version20240423AsInt)
+            .HasApiVersion(ApiVersions.Version20230101AsInt);
+        app.UseActivityLogWithB2CSupport().WithApiVersionSet(activityLogApiVersionSet)
+            .HasApiVersion(ApiVersions.Version20240515AsInt);
 
-app.Run();
 
-public partial class Program
-{
+        if (args.Contains("--swagger"))
+        {
+            var swaggerProvider = app.Services.GetRequiredService<ISwaggerProvider>();
+            var swagger = swaggerProvider.GetSwagger(ApiVersions.Version20240515);
+
+            File.WriteAllText(
+                Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, "contracts.yaml"),
+                swagger.SerializeAsYaml(OpenApiSpecVersion.OpenApi3_0)
```

Comment:
Redoc is able to convert json to yaml by itself. At least that is what it does in the init container. No need to do it here.

Reply:
the OpenApiDocument type makes you serialize as yaml or json, no need to serialize json and make it yaml in the cli